components/org.wso2.carbon.identity.api.server.notification.template/org.wso2.carbon.identity.rest.api.server.notification.template.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/notification/template/v1/util/Util.java

```diff
@@ -0,0 +1,350 @@
+        templateWithID.setBody(internalTemplate.getBody());
+        templateWithID.setFooter(internalTemplate.getFooter());
+        return templateWithID;
+    }
+
+    /**
+     * Builds SMSTemplateWithID using NotificationTemplate.
+     *
+     * @param internalTemplate NotificationTemplate object.
+     * @return SMSTemplateWithID object.
+     */
+    public static SMSTemplateWithID buildSMSTemplateWithID(NotificationTemplate internalTemplate) {
+
+        SMSTemplateWithID templateWithID = new SMSTemplateWithID();
+        templateWithID.setLocale(internalTemplate.getLocale());
+        templateWithID.setBody(internalTemplate.getBody());
+        return templateWithID;
+    }
+
+    /**
+     * Builds NotificationTemplate object using SMSTemplateWithID object.
+     *
+     * @param templateTypeId    Template type ID.
+     * @param smsTemplateWithID SMSTemplateWithID Object.
+     * @return NotificationTemplate object built using provided values.
+     */
+    public static NotificationTemplate buildNotificationTemplateWithSMSTemplateWithID(String templateTypeId,
+                                                                               SMSTemplateWithID smsTemplateWithID) {
+
+        String templateTypeDisplayName = Util.decodeTemplateTypeId(templateTypeId);
+        NotificationTemplate notificationTemplate = new NotificationTemplate();
+        notificationTemplate.setNotificationChannel(Constants.NOTIFICATION_CHANNEL_SMS);
+        notificationTemplate.setLocale(smsTemplateWithID.getLocale());
+        notificationTemplate.setBody(smsTemplateWithID.getBody());
+        notificationTemplate.setDisplayName(templateTypeDisplayName);
+        notificationTemplate.setLocale(smsTemplateWithID.getLocale());
+        notificationTemplate.setType(I18nEmailUtil.getNormalizedName(templateTypeDisplayName));
+        return notificationTemplate;
+    }
+
+    /**
+     * Builds NotificationTemplate object using EmailTemplateWithID object.
+     *
+     * @param templateTypeId      Template type ID.
+     * @param emailTemplateWithID EmailTemplateWithID Object.
+     * @return NotificationTemplate object built using provided values.
+     */
+    public static NotificationTemplate buildNotificationTemplateWithEmailTemplateWithID(String templateTypeId,
+                                                                           EmailTemplateWithID emailTemplateWithID) {
+
+        String templateTypeDisplayName = Util.decodeTemplateTypeId(templateTypeId);
+        NotificationTemplate notificationTemplate = new NotificationTemplate();
+        notificationTemplate.setNotificationChannel(Constants.NOTIFICATION_CHANNEL_EMAIL);
+        notificationTemplate.setLocale(emailTemplateWithID.getLocale());
+        notificationTemplate.setBody(emailTemplateWithID.getBody());
+        notificationTemplate.setDisplayName(templateTypeDisplayName);
+        notificationTemplate.setLocale(emailTemplateWithID.getLocale());
```

Comment:
duplicate setting  locale. check for similar issues.

Reply:
addressed with https://github.com/wso2/identity-api-server/pull/629/commits/d4df259f15fdad63db051f2ed29160d53e091aec